subprojects/core/src/main/java/org/gradle/invocation/DefaultGradle.java

```diff
@@ -245,9 +247,6 @@ public ClassLoaderScope baseProjectClassLoaderScope() {
 
     @Override
     public void setBaseProjectClassLoaderScope(ClassLoaderScope classLoaderScope) {
-        if (classLoaderScope == null) {
```

Comment:
❓ What is the motivation for removing this assertion?

Reply:
According to the nullability annotations, this parameter is never null. 

In general, we assume non-nullability unless otherwise annotated. 

I went through all usages of this method and verified that there was indeed nobody calling this with a null parameter, so I removed the check.

Reply:
Ok. Would passing null here fail the build then?

Reply:
I can add the null check back. 

Should I remove all the other nullability corrections I made in this PR as well? There were many IDE warnings in this class related to nullability -- and specifically for this case if the IDE warns that this field is _never_ null.

I was under the impression that it is good practice to address these nullability warnings as we come across them, but I can refrain from touching nullability in cortex PRs if you think they should be avoided.

Reply:
> I was under the impression that it is good practice to address these nullability warnings 

It is. And please don't get me wrong, I do appreciate the attitude.

Adding obviously missing annotations are fine.

Removing nullability checks are more distracting at the moment since we don't have good tooling so I'd rather keep them in separate PRs.